src/v/pandaproxy/rest/proxy.cc

```diff
@@ -22,51 +29,93 @@ namespace pandaproxy::rest {
 
 using server = ctx_server<proxy>;
 
+template<typename Handler>
+auto wrap(Handler h) {
+    return [h{std::move(h)}](
+             server::request_t rq,
+             server::reply_t rp) -> ss::future<server::reply_t> {
+        rq.authn_method = rq.service().config().authn_method(
+          rq.req->get_listener_idx());
+
+        if (rq.authn_method == config::rest_authn_method::http_basic) {
+            // Will throw if auth fails
+            // The catch statements in reply.h will do error handling
+            auto auth_result = rq.service().authenticator().authenticate(
+              *rq.req);
+            auth_result.pass();
```

Comment:
Should this be `.require_authenticated()`?

Reply:
Took a look at this.
`.authenticate()` will throw 400 & 401 if user-pass validation fails
`auth_result.require_authenticated()` For the PP, this will throw 403 if the user enable HTTP Basic AuthN but did not give the authroziation header.

Putting it in.